src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2428,17 +2443,128 @@ private void tryAddSecurityProvider() {
     // CS-SUPPRESS-SINGLE: RegexpSingleline get Resource Sharing Extensions
     @Override
     public void loadExtensions(ExtensionLoader loader) {
-
-        if (settings != null
-            && settings.getAsBoolean(
+        if (settings == null
+            || !settings.getAsBoolean(
                 ConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED,
                 ConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED_DEFAULT
             )) {
-            // load all resource-sharing extensions
-            resourceSharingExtensions.addAll(new HashSet<>(loader.loadExtensions(ResourceSharingExtension.class)));
-            resourcePluginInfo.setResourceSharingExtensions(resourceSharingExtensions);
+            return;
+        }
+
+        // discover & register extensions and their types
+        Set<ResourceSharingExtension> exts = new HashSet<>(loader.loadExtensions(ResourceSharingExtension.class));
+        resourcePluginInfo.setResourceSharingExtensions(exts);
+
+        for (var ext : exts) {
+            URL url = ext.getClass().getClassLoader().getResource("resource-action-groups.yml");
```

Comment:
Should we create a class like [ConfigHelper](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/support/ConfigHelper.java) for this?